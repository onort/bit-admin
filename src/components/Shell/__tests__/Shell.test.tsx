import React from "react"
import { mount, shallow } from "enzyme"
import { MemoryRouter as Router } from "react-router-dom"

import { AppBar, AppMain, Shell, Sidebar } from "../../"

describe("<Shell />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Router>
        <Shell />
      </Router>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have AppBar component", () => {
    const wrapper = mount(
      <Router>
        <Shell />
      </Router>
    )
    expect(wrapper.find(AppBar).exists()).toBe(true)
    wrapper.unmount()
  })

  it("should have AppMain component", () => {
    const wrapper = mount(
      <Router>
        <Shell />
      </Router>
    )
    expect(wrapper.find(AppMain).exists()).toBe(true)
    wrapper.unmount()
  })

  it("should have Sidebar component", () => {
    const wrapper = mount(
      <Router>
        <Shell />
      </Router>
    )
    expect(wrapper.find(Sidebar).exists()).toBe(true)
    wrapper.unmount()
  })

  it("should toggle sidebar on menuIcon click", () => {
    const wrapper = mount(
      <Router>
        <Shell />
      </Router>
    )
    expect(wrapper.find(Shell).state().sidebarOpen).toBe(true)
    expect(wrapper.find(".closed").exists()).toBe(false)
    wrapper.find(".menuIcon").simulate("click")
    expect(wrapper.find(Shell).state().sidebarOpen).toBe(false)
    expect(wrapper.find(".closed").exists()).toBe(true)
    wrapper.unmount()
  })
})
