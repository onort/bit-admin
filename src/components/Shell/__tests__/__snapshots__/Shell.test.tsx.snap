// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shell /> should match snapshot 1`] = `"<div class=\\"container\\"><header class=\\"container\\"><div class=\\"appBar\\"><div class=\\"brandContainer\\"><span class=\\"menuIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"brand\\">Bmat Bit Admin</span></div></div></header><div class=\\"applicationContainer\\"><nav class=\\"container\\"><a class=\\"item\\" href=\\"/add\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></span><span class=\\"itemTitle\\">Add</span></a><a class=\\"item\\" href=\\"/view\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z\\"></path></svg></span><span class=\\"itemTitle\\">View</span></a><a class=\\"item logout\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg></span><span class=\\"itemTitle\\">Logout</span></a></nav><main class=\\"container\\"></main></div></div>"`;
